fairlearn/datasets/_synthetic.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+from collections import OrderedDict
+from itertools import product
+
+import numpy as np
+import pandas as pd
+from sklearn.datasets import make_classification
+from sklearn.utils import check_random_state
+
+
+class SensitiveFeatureGroupConfig:
+
+    def __init__(self, group_dict, **classification_kwargs):
+        """ `group_dict` is a dict assigning to a group in each sensitive
+        feature category. E.g.,::
+            {
+                'gender': 'Other',
+                'age': '50-60',
+            }
+        `classification_kwargs` is the set of kwargs that will be passed on to
+        make_classification.
```

Comment:
It may be preferable to have a description of the class, then an example and the parameters section.